Add OptiPlex profiles

DaniilKl opened the pull request.

Commit:
platform-configs: optiplex: add correct dmidecode values

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

m-iwanicki commented on platform-configs/optiplex-9010.robot:
```diff
@@ -1,2 +1,6 @@
+
+${DMIDECODE_PRODUCT_NAME}=      OptiPlex 9010
+${DMIDECODE_SERIAL_NUMBER}=     123456789
+${DMIDECODE_MANUFACTURER}=      Dell Inc.
```
add to `optiplex-common.robot`?

DaniilKl replied:
8b79b6bab5bc0906ae8cf59664b52185080a17bf

Commit:
platform-configs: optiplex-common: add common DTS exports

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

dts: profile: optiplex-[79]010: add Initial deployment profiles

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

dts: profiles: add OptiPlex UEFI Update profiles

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

platform-configs: optiplex-common: fix pre-commit

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

platform-configs: optiplex: delete DTS TEST_*_MIGRATED vars

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

platform-configs: optiplex: delete TEST_IS_COREBOOT

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

Commit:
platform-configs: optiplex: fix old firmware menu configs

Signed-off-by: Daniil Klimuk <daniil.klimuk@3mdeb.com>

m-iwanicki commented on platform-configs/include/optiplex-common.robot:
```diff
@@ -92,13 +92,27 @@ ${BASE_PORT_RAMSTAGE_SUPPORT}=                  ${TRUE}
+&{DTS_TEST_EXPORTS_PER_WORKFLOW_BASE}=
+...                                                 UEFI Update=&{{ {"TEST_IS_COREBOOT": "true"} }}
+...                                                 UEFI Update=&{{ {"TEST_ME_DISABLED": "false"} }}
+...                                                 UEFI Update=&{{ {"TEST_ME_HAP_DISABLED": "true"} }}
```
Only last value is taken into consideration, it overwrites previous ones e.g.:

```
&{DICT}=    a=1    a=2    a=3
```

will result in `DICT` having one key `a` with value `3`

DaniilKl replied:
Do you mean that I am overwriting the the `UEFI Update`?

m-iwanicki replied:
Yes

DaniilKl replied:
Good catch, thank you. https://github.com/Dasharo/open-source-firmware-validation/compare/9a19c48956386cd278adab75b54737783ba8fa54..7dd7103a3d23a83eaa4690134d56bd901145bac5

DaniilKl replied:
https://github.com/Dasharo/open-source-firmware-validation/compare/7dd7103a3d23a83eaa4690134d56bd901145bac5..5c2d37cec5871bdcc66ff1963361ec1cd0d59841

m-iwanicki commented on platform-configs/include/optiplex-common.robot:
```diff
@@ -92,13 +92,27 @@ ${BASE_PORT_RAMSTAGE_SUPPORT}=                  ${TRUE}
+...                                             TEST_ACM_MIGRATED=false
+...                                             TEST_SOUND_CARD_PRESENT=false
+...                                             TEST_MEI_CONF_PRESENT=false
+&{DTS_TEST_EXPORTS_PER_WORKFLOW_BASE}=
```
```suggestion
&{DTS_TEST_EXPORTS_PER_WORKFLOW}=
...    &{DTS_TEST_EXPORTS_PER_WORKFLOW_BASE}
...    UEFI Update=&{{ {"TEST_IS_COREBOOT": "true", "TEST_ME_HAP_DISABLED": "true"} }}
```
?

m-iwanicki replied:
https://github.com/Dasharo/open-source-firmware-validation/blob/f99e4b605fa8522b71bb2340ebd11beafd6192f2/platform-configs/odroid-h4-plus.robot#L126-L128

DaniilKl replied:
Good catch, thank you. https://github.com/Dasharo/open-source-firmware-validation/compare/9a19c48956386cd278adab75b54737783ba8fa54..7dd7103a3d23a83eaa4690134d56bd901145bac5

m-iwanicki commented on platform-configs/include/optiplex-common.robot:
```diff
@@ -92,13 +92,26 @@ ${BASE_PORT_RAMSTAGE_SUPPORT}=                  ${TRUE}
+&{DTS_TEST_EXPORTS_PER_WORKFLOW}=
+...                                             &{DTS_TEST_EXPORTS_PER_WORKFLOW_BASE}
+...                                             UEFI Update=&{{ {"TEST_IS_COREBOOT": "true", "TEST_ME_DISABLED": "false", "TEST_ME_HAP_DISABLED": "true"} }}
+...                                             Initial Deployment=&{{ {"TEST_IS_COREBOOT": "false"} }}
```
You can remove `Initial Deployment` as default value of `TEST_IS_COREBOOT` is empty/false. Or you could add it to `DTS_TEST_EXPORTS_PER_WORKFLOW_BASE` in `default.robot` if you want this export for every platform.

m-iwanicki committed:
dts-e2e: fix platform/workflow/release parsing from test name

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts-e2e: Force profile verification based on platform-config variable

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts-e2e: add comments to MSI Heads transition profiles

Either transition didn't work or couldn't be verified

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>
